sklearn/feature_selection/from_model.py

```diff
@@ -123,25 +129,52 @@ class SelectFromModel(BaseEstimator, SelectorMixin, MetaEstimatorMixin):
     threshold_ : float
         The threshold value used for feature selection.
     """
-    def __init__(self, estimator, threshold=None, prefit=False, norm_order=1):
+    def __init__(self, estimator, threshold=None, prefit=False,
+                 max_features=None, norm_order=1):
         self.estimator = estimator
         self.threshold = threshold
         self.prefit = prefit
+        self.max_features = max_features
         self.norm_order = norm_order
 
+    def _check_max_features(self, X, max_features):
+        if self.max_features is None or self.max_features == 'all':
```

Comment:
is there a reason to have both 'all' and None available? Can't we just allow the default value, either 'all' or None, to yield this behaviour?

Reply:
Agreed. Removed support for "all".